frontend/src/app/features/member/member.service.ts

```diff
@@ -16,46 +15,43 @@ export class MemberService {
 
   getAllMembers(): Observable<MemberModel[]> {
     return this.httpClient.get<MemberModel[]>(this.API_URL)
-      .pipe(mergeMap((members) => from(members)), this.formatDate(), toArray());
+      .pipe(map((dtos) => dtos.map((dto) => this.mapToModel(dto))));
   }
 
-
   getMemberById(id: number): Observable<MemberModel> {
     return this.httpClient.get<MemberModel>(`${this.API_URL}/${id}`)
-      .pipe(this.formatDate());
+      .pipe(map((dto) => this.mapToModel(dto)));
   }
 
   addMember(member: MemberModel): Observable<MemberModel> {
     return this.httpClient.post<MemberModel>(this.API_URL, this.toDto(member))
-      .pipe(this.formatDate());
+      .pipe(map((dto) => this.mapToModel(dto)));
   }
 
   updateMember(id: number, member: MemberModel): Observable<MemberModel> {
     return this.httpClient.put<MemberModel>(`${this.API_URL}/${id}`, this.toDto(member))
-      .pipe(this.formatDate());
+      .pipe(map((dto) => this.mapToModel(dto)));
   }
 
-  /*
-   *This is a temporary solution to read the Date we receive from backend
-   */
-  formatDate() {
-    return tap((member: MemberModel) => {
-      member.birthDate = DateTime.fromISO(member.birthDate.toString());
-      if (member.dateOfHire) {
-        member.dateOfHire = DateTime.fromISO(member.dateOfHire.toString());
-      }
-    });
+  private mapToModel(dto: any): MemberModel {
```

Comment:
```suggestion
  private fromDto(dto: any): MemberModel {
```
Why no use `fromDto` as the name? This keeps it consistent with the `toDto()` method we already have and with the backend aswell.